### ISSUE TITLE
Support ssh2 for PHP 8.0

### ISSUE THREAD

**Issue by evs-xsarus** [mlocati/docker-php-extension-installer#282]
Currently, ssh2 can't be installed on PHP 8.0.

NB This will probably be fixed 'magically' in time according to this PR: https://github.com/php/pecl-networking-ssh2/pull/44

From the install log:

```log
### INSTALLING REMOTE MODULE ssh2 ###
  (installing version 1.2)
Downloading (100%)          Downloading (connecting...)
+-----------------------------------+------+
| Package name                      | ssh2 |
| Package version (current release) | 1.2  |
| Package status                    | beta |
+-----------------------------------+------+
/The following error(s) happened: make failed
1: phpize
2: Configuring for:
2: PHP Api Version:         20200930
2: Zend Module Api No:      20200930
2: Zend Extension Api No:   420200930
1: /tmp/pickle.tmp/ssh2/ssh2-1.2/configure  
2: checking for grep that handles long lines and -e... /bin/grep
2: checking for egrep... /bin/grep -E
2: checking for a sed that does not truncate output... /bin/sed
2: checking for pkg-config... /usr/bin/pkg-config
2: checking pkg-config is at least version 0.9.0... yes
2: checking for cc... cc
2: checking whether the C compiler works... yes
2: checking for C compiler default output file name... a.out
2: checking for suffix of executables...
2: checking whether we are cross compiling... no
2: checking for suffix of object files... o
2: checking whether we are using the GNU C compiler... yes
2: checking whether cc accepts -g... yes
2: checking for cc option to accept ISO C89... none needed
2: checking how to run the C preprocessor... cc -E
2: checking for icc... no
2: checking for suncc... no
2: checking for system library directory... lib
2: checking if compiler supports -R... no
2: checking if compiler supports -Wl,-rpath,... yes
2: checking build system type... x86_64-pc-linux-gnu
2: checking host system type... x86_64-pc-linux-gnu
2: checking target system type... x86_64-pc-linux-gnu
2: checking for PHP prefix... /usr/local
2: checking for PHP includes... -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib
2: checking for PHP extension directory... /usr/local/lib/php/extensions/no-debug-non-zts-20200930
2: checking for PHP installed headers prefix... /usr/local/include/php
2: checking if debug is enabled... no
2: checking if zts is enabled... no
2: checking for gawk... no
2: checking for nawk... nawk
2: checking if nawk is broken... no
2: checking for ssh2 support... yes, shared
2: checking for ssh2 files in default path... found in /usr
2: checking for libssh2_session_hostkey in -lssh2... yes
2: checking for libssh2_agent_init in -lssh2... yes
2: checking for libssh2_session_set_timeout in -lssh2... yes
2: checking for a sed that does not truncate output... /bin/sed
2: checking for ld used by cc... /usr/bin/ld
2: checking if the linker (/usr/bin/ld) is GNU ld... yes
2: checking for /usr/bin/ld option to reload object files... -r
2: checking for BSD-compatible nm... /usr/bin/nm -B
2: checking whether ln -s works... yes
2: checking how to recognize dependent libraries... pass_all
2: checking for ANSI C header files... yes
2: checking for sys/types.h... yes
2: checking for sys/stat.h... yes
2: checking for stdlib.h... yes
2: checking for string.h... yes
2: checking for memory.h... yes
2: checking for strings.h... yes
2: checking for inttypes.h... yes
2: checking for stdint.h... yes
2: checking for unistd.h... yes
2: checking dlfcn.h usability... yes
2: checking dlfcn.h presence... yes
2: checking for dlfcn.h... yes
2: checking the maximum length of command line arguments... 1572864
2: checking command to parse /usr/bin/nm -B output from cc object... ok
2: checking for objdir... .libs
2: checking for ar... ar
2: checking for ranlib... ranlib
2: checking for strip... strip
2: checking if cc supports -fno-rtti -fno-exceptions... no
2: checking for cc option to produce PIC... -fPIC
2: checking if cc PIC flag -fPIC works... yes
2: checking if cc static flag -static works... yes
2: checking if cc supports -c -o file.o... yes
2: checking whether the cc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
2: checking whether -lc should be explicitly linked in... no
2: checking dynamic linker characteristics... GNU/Linux ld.so
2: checking how to hardcode library paths into programs... immediate
2: checking whether stripping libraries is possible... yes
2: checking if libtool supports shared libraries... yes
2: checking whether to build shared libraries... yes
2: checking whether to build static libraries... no
2: 
2: creating libtool
2: appending configuration tag "CXX" to libtool
2: configure: patching config.h.in
2: configure: creating ./config.status
2: config.status: creating config.h
1: make
2: /bin/bash /tmp/pickle.tmp/pickle-325b9901a0667b1f9eb53063ee1dfb10ab4517dc/libtool --mode=compile cc -I. -I/tmp/pickle.tmp/ssh2/ssh2-1.2 -I/tmp/pickle.tmp/pickle-325b9901a0667b1f9eb53063ee1dfb10ab4517dc/include -I/tmp/pickle.tmp/pickle-325b9901a0667b1f9eb53063ee1dfb10ab4517dc/main -I/tmp/pickle.tmp/ssh2/ssh2-1.2 -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O2    -c /tmp/pickle.tmp/ssh2/ssh2-1.2/ssh2.c -o ssh2.lo
2: mkdir .libs
2:  cc -I. -I/tmp/pickle.tmp/ssh2/ssh2-1.2 -I/tmp/pickle.tmp/pickle-325b9901a0667b1f9eb53063ee1dfb10ab4517dc/include -I/tmp/pickle.tmp/pickle-325b9901a0667b1f9eb53063ee1dfb10ab4517dc/main -I/tmp/pickle.tmp/ssh2/ssh2-1.2 -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /tmp/pickle.tmp/ssh2/ssh2-1.2/ssh2.c  -fPIC -DPIC -o .libs/ssh2.o
2: /tmp/pickle.tmp/ssh2/ssh2-1.2/ssh2.c:49:2: warning: implicit declaration of function 'ZEND_ARG_PASS_INFO'; did you mean 'ZEND_ARG_OBJ_INFO'? [-Wimplicit-function-declaration]
2:   ZEND_ARG_PASS_INFO(1)
2:   ^~~~~~~~~~~~~~~~~~
2:   ZEND_ARG_OBJ_INFO
2: /tmp/pickle.tmp/ssh2/ssh2-1.2/ssh2.c:49:2: warning: initialization of 'const char *' from 'int' makes pointer from integer without a cast [-Wint-conversion]
2: /tmp/pickle.tmp/ssh2/ssh2-1.2/ssh2.c:49:2: note: (near initialization for 'php_ssh2_first_arg_force_ref[1].name')
2: /tmp/pickle.tmp/ssh2/ssh2-1.2/ssh2.c:49:2: error: initializer element is not constant
2: /tmp/pickle.tmp/ssh2/ssh2-1.2/ssh2.c:49:2: note: (near initialization for 'php_ssh2_first_arg_force_ref[1].name')
2: /tmp/pickle.tmp/ssh2/ssh2-1.2/ssh2.c: In function 'php_ssh2_debug_cb':
2: /tmp/pickle.tmp/ssh2/ssh2-1.2/ssh2.c:103:17: warning: implicit declaration of function 'call_user_function_ex'; did you mean 'call_user_function'? [-Wimplicit-function-declaration]
2:   if (FAILURE == call_user_function_ex(NULL, NULL, data->disconnect_cb, NULL, 3, args, 0, NULL)) {
2:                  ^~~~~~~~~~~~~~~~~~~~~
2:                  call_user_function
2: /tmp/pickle.tmp/ssh2/ssh2-1.2/ssh2.c: In function 'zif_ssh2_disconnect':
2: /tmp/pickle.tmp/ssh2/ssh2-1.2/ssh2.c:443:6: error: void value not ignored as it ought to be
2:   if (zend_list_close(Z_RES_P(zsession)) != SUCCESS) {
2:       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2: /tmp/pickle.tmp/ssh2/ssh2-1.2/ssh2.c:443:41: error: invalid operands to binary != (have 'const zend_internal_arg_info[1]' {aka 'const struct _zend_internal_arg_info[1]'} and 'int')
2:   if (zend_list_close(Z_RES_P(zsession)) != SUCCESS) {
2:       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^~
```

**Comment by mlocati** [comment]
Yep, that's why it's not listed as supported in the [README](https://github.com/mlocati/docker-php-extension-installer#supported-php-extensions).

We need that they publish a new release (it seems they are actively working on it).

**Comment by mlocati** [comment]
I manually implemented a fix because the people managing the ssh2 haven't published an official release yet

**Comment by mlocati** [comment]
We now install the official release of ssh2, not a development version (they released version 1.3 with PHP 8 support).
